src/Drivers/Xendit/Xendit.php

```diff
@@ -0,0 +1,222 @@
+                       ?? Request::input('id')
+                       ?? Request::input('external_id');
+        }
+
+        if (empty($invoiceId)) {
+            throw new InvalidPaymentException('Invoice ID not found for verification.');
+        }
+
+        try {
+            $response = $this->client->get("v2/invoices/{$invoiceId}");
+            $result = json_decode((string) $response->getBody(), true);
+
+            if (!isset($result['id'])) {
+                throw new InvalidPaymentException('Invalid invoice response.');
+            }
+
+            // Check invoice status
+            if ($result['status'] !== 'SETTLED') {
+                throw new InvalidPaymentException(
+                    'Invoice not paid. Status: ' . ($result['status'] ?? 'Unknown')
+                );
+            }
+
+            // Verify amount matches
+            if ($result['amount'] != $this->invoice->getAmount()) {
+                throw new InvalidPaymentException('Invoice amount mismatch.');
+            }
+
+            return $this->createReceipt($result);
+        } catch (\Exception $e) {
+            if ($e instanceof InvalidPaymentException) {
+                throw $e;
+            }
+            throw new InvalidPaymentException('Xendit invoice verification failed: ' . $e->getMessage());
+        }
+    }
+
+    protected function createReceipt(array $invoiceData): Receipt
+    {
+        $receipt = new Receipt('xendit', $invoiceData['id']);
+        
+        // Add additional invoice details if available
+        if (isset($invoiceData['external_id'])) {
+            $receipt->detail('external_id', $invoiceData['external_id']);
+        }
+
+        if (isset($invoiceData['payment_method'])) {
+            $receipt->detail('payment_method', $invoiceData['payment_method']);
+        }
+
+        if (isset($invoiceData['payment_channel'])) {
+            $receipt->detail('payment_channel', $invoiceData['payment_channel']);
+        }
+
+        if (isset($invoiceData['payment_destination'])) {
+            $receipt->detail('payment_destination', $invoiceData['payment_destination']);
+        }
+
+        if (isset($invoiceData['paid_at'])) {
+            $receipt->detail('paid_at', $invoiceData['paid_at']);
+        }
+
+        if (isset($invoiceData['payment_id'])) {
+            $receipt->detail('payment_id', $invoiceData['payment_id']);
+        }
+
+        return $receipt;
+    }
+
+    /**
+     * Get invoice details
+     */
+    public function getInvoice(string $invoiceId = null): array
+    {
+        try {
+            $response = $this->client->get("v2/invoices/{$invoiceId}");
+            return json_decode((string) $response->getBody(), true);
+        } catch (\Exception $e) {
+            throw new InvalidPaymentException('Failed to get invoice: ' . $e->getMessage());
+        }
+    }
+
+    /**
+     * Expire an invoice
+     */
+    public function expireInvoice(string $invoiceId): bool
```

Comment:
is this method does a lookup to check if the invoice is expired ? or it marks it as expired ? I can't understand the functionality here because this function returns a boolean!

Also it is not a good practice to return `false` on exception! the exception should be catched and handled by the caller.